File: csharp/Easy.cs
using System;
using System.Data.SqlClient;
using System.Web;

public class Easy
{
    // SQL Injection
    public void Login(HttpRequest req, SqlConnection conn)
    {
        string user = req.QueryString["user"];
        string pass = req.QueryString["pass"];
        string query = "SELECT * FROM Users WHERE Username = '" + user + "' AND Password = '" + pass + "'";
        SqlCommand cmd = new SqlCommand(query, conn);
        var reader = cmd.ExecuteReader(); // Vulnerable to SQL Injection
    }

    // Cross-Site Scripting (XSS)
    public void Greet(HttpResponse res, HttpRequest req)
    {
        string name = req.QueryString["name"];
        res.Write("Hello " + name + ""); // No output encoding - XSS vulnerable
    }

    // Path Traversal
    public void ReadFile(HttpResponse res, HttpRequest req)
    {
        string filename = req.QueryString["file"];
        string content = System.IO.File.ReadAllText(@"C:\temp\" + filename); // No validation
        res.Write(content);
    }
}

File: go/easy.go
package main

import (
	"database/sql"
	"fmt"
	"net/http"
	"os"
)

func main() {

	http.HandleFunc("/login", func(w http.ResponseWriter, r *http.Request) {
		user := r.URL.Query().Get("user")
		pass := r.URL.Query().Get("pass")
		// vulnerable: user input directly in query
		q := fmt.Sprintf("SELECT * FROM users WHERE user='%s' AND pass='%s'", user, pass)
		db.Query(q)
	})


	http.HandleFunc("/show", func(w http.ResponseWriter, r *http.Request) {
		file := r.URL.Query().Get("file")
		// vulnerable: no input validation
		data, _ := os.ReadFile("/tmp/" + file)
		w.Write(data)
	})


	http.HandleFunc("/run", func(w http.ResponseWriter, r *http.Request) {
		name := r.URL.Query().Get("person")
		_ = os.system("echo Hello " + name)
	})
}
var db *sql.DB
